shop4All-backend/src/main/java/com/example/shop4All_backend/services/ProductService.java

```diff
@@ -19,8 +26,92 @@ public class ProductService {
 
     //add a product
     public Product addNewProduct(Product product) {
+        validateCredentials(product);
```

Comment:
I think you can make this method return a bool and, after this line, you can do a check if the credentials are validated. If so, do the addNewProduct logic, else throw an Exception.